src/input_common/gcadapter/gc_adapter.cpp

```diff
@@ -283,7 +285,88 @@ void Adapter::Reset() {
+        {Settings::NativeButton::A, PadButton::PAD_BUTTON_A},
+        {Settings::NativeButton::B, PadButton::PAD_BUTTON_B},
+        {Settings::NativeButton::X, PadButton::PAD_BUTTON_X},
+        {Settings::NativeButton::Y, PadButton::PAD_BUTTON_Y},
+        {Settings::NativeButton::Plus, PadButton::PAD_BUTTON_START},
+        {Settings::NativeButton::DLeft, PadButton::PAD_BUTTON_LEFT},
+        {Settings::NativeButton::DUp, PadButton::PAD_BUTTON_UP},
+        {Settings::NativeButton::DRight, PadButton::PAD_BUTTON_RIGHT},
+        {Settings::NativeButton::DDown, PadButton::PAD_BUTTON_DOWN},
+        {Settings::NativeButton::SL, PadButton::PAD_TRIGGER_L},
+        {Settings::NativeButton::SR, PadButton::PAD_TRIGGER_R},
+        {Settings::NativeButton::R, PadButton::PAD_TRIGGER_Z},
+    };
+    if (!params.Has("port")) {
+        return {};
+    }
+
+    InputCommon::ButtonMapping mapping{};
+    for (const auto& [switch_button, gcadapter_button] : switch_to_gcadapter_button) {
+        Common::ParamPackage button_params({{"engine", "gcpad"}});
+        button_params.Set("port", params.Get("port", 0));
+        button_params.Set("button", static_cast<int>(gcadapter_button));
+        mapping[switch_button] = button_params;
+    }
+
+    // Add the missing bindings for ZL/ZR
+    std::unordered_map<Settings::NativeButton::Values, PadAxes> switch_to_gcadapter_axis = {
+        {Settings::NativeButton::ZL, PadAxes::TriggerLeft},
+        {Settings::NativeButton::ZR, PadAxes::TriggerRight},
+    };
+    for (const auto& [switch_button, gcadapter_axis] : switch_to_gcadapter_axis) {
+        Common::ParamPackage button_params({{"engine", "gcpad"}});
+        button_params.Set("port", params.Get("port", 0));
+        button_params.Set("button", static_cast<int>(PadButton::PAD_STICK));
+        button_params.Set("axis", static_cast<int>(gcadapter_axis));
+        mapping[switch_button] = button_params;
+    }
+    return mapping;
+}
+
+InputCommon::AnalogMapping Adapter::GetAnalogMappingForDevice(const Common::ParamPackage& params) {
+    if (!params.Has("port")) {
+        return {};
+    }
+
+    InputCommon::AnalogMapping mapping = {};
+    Common::ParamPackage left_analog_params;
+    left_analog_params.Set("engine", "gcpad");
+    left_analog_params.Set("port", params.Get("port", 0));
+    left_analog_params.Set("axis_x", static_cast<int>(PadAxes::StickX));
+    left_analog_params.Set("axis_y", static_cast<int>(PadAxes::StickY));
+    mapping[Settings::NativeAnalog::LStick] = left_analog_params;
+    Common::ParamPackage right_analog_params;
+    right_analog_params.Set("engine", "gcpad");
+    right_analog_params.Set("port", params.Get("port", 0));
+    right_analog_params.Set("axis_x", static_cast<int>(PadAxes::SubstickX));
+    right_analog_params.Set("axis_y", static_cast<int>(PadAxes::SubstickY));
+    mapping[Settings::NativeAnalog::RStick] = right_analog_params;
```

Comment:
```suggestion
    mapping.insert_or_assign(Settings::NativeAnalog::RStick, std::move(right_analog_params));
```

Ditto for the assignment for the left analog version as well.